### ISSUE TITLE
can't apply schema/diff

### ISSUE THREAD

**Issue by jbsiddall** [directus/directus#18286]
### Describe the Bug

We store directus schema snapshots in our git repo so we can automatically deploy schema changes in staging and prod.
Sometimes when we hit directus's endpoint schema/diff, it produces a diff file that can't be consumed by schema/apply.

This is a huge problem for us because in dev environment we just wipe directus's database and now diff/apply works, but in production we litterally couldn't update our schema.

### To Reproduce

* start with clean empty directus at version 9.23.4. 
* in folder somewhere unzip the attached example.zip and you'll find 3 files.
* hit the directus endpoint schema/diff and schema/apply with first directus schema directus-snapshot-in-our-directus-server.json. This is expected to work.
* hit the directus endpoint schema/diff with schema directus-snapshot-in-git.json. A diff will be generated that is expected to look like the file directus-diff.json. Please note that `rhs: null` value in the diff which appears to cause the errors.
* attempt to hit endpoint schema/diff with either directus-snapshot-in-git.json or the diff you generated (they should be the same). I'd expected this to fail with an error message:
```
{"errors":[{"message":"\"diff.collections[0].diff[0].lhs\" must be of type object","extensions":{"code":"INVALID_PAYLOAD","stack":"Error: \"diff.collections[0].diff[0].lhs\" must be of type object\n    at validateApplyDiff (/directus/dist/utils/validate-diff.js:54:15)\n    at SchemaService.apply (/directus/dist/services/schema.js:33:52)\n    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at async /directus/dist/controllers/schema.js:25:5"}}]}
```

[example.zip](https://github.com/directus/directus/files/11296929/example.zip)

### Directus Version

9.23.4

### Hosting Strategy

Self-Hosted (Docker Image)

**Comment by azrikahar** [comment]
This should be a duplicate of #17698 and was fixed in `9.25.0` by #18048 👍

**Comment by jbsiddall** [comment]
thank you!
